### PR TITLE
Add get update methods

### PR DESCRIPTION
**Pull request by ruuushhh**

### CODE REVIEW

File: credit_card_charge_restlet/create_credit_card_charge.js [hunk 1]
```diff
@@ -30,6 +17,184 @@ function (record) {
...
+    function doGet(requestParams) {
+        try {
+            // Load the credit card charge record in dynamic mode
+            var creditCardCharge = record.load({
+                type: record.Type.CREDIT_CARD_CHARGE,
+                id: requestParams.internalId,
+                isDynamic: true
+            });
+    
+            // Prepare the response object with all relevant fields
```
Comment (ashwin1111) [inline credit_card_charge_restlet/create_credit_card_charge.js]: why not just throw whole dump? we might miss internal fields if we construct things. When you have whole evidence why filter and send

File: credit_card_charge_restlet/create_credit_card_charge.js [hunk 2]
```diff
@@ -30,6 +17,184 @@ function (record) {
...
+    // Function called upon sending a PUT request to update a record
+    function doPut(requestBody) {
+        try {
+            var creditCardCharge = record.load({
+                type: record.Type.CREDIT_CARD_CHARGE,
+                id: requestBody.internalId,
+                isDynamic: true
+            });
+
+            creditCardCharge.setValue({ fieldId: 'memo', value: requestBody.memo });
```
Comment (ashwin1111) [inline credit_card_charge_restlet/create_credit_card_charge.js]: let's add all fields that we have for post()

File: credit_card_charge_restlet/create_credit_card_charge.js [hunk 3]
```diff
@@ -30,6 +17,184 @@ function (record) {
...
+
+            // First, remove existing expense lines
+            var expenseLineCount = creditCardCharge.getLineCount({ sublistId: 'expense' });
+            for (var i = expenseLineCount - 1; i >= 0; i--) {
+                creditCardCharge.removeLine({ sublistId: 'expense', line: i });
+            }
```
Comment (ashwin1111) [inline credit_card_charge_restlet/create_credit_card_charge.js:61]: what's the reason for lines removal and addition and not update it directly?

File: credit_card_charge_restlet/create_credit_card_charge.js (hunk 1 above)
Comment (ruuushhh) [reply credit_card_charge_restlet/create_credit_card_charge.js]: the response we get from netsuite is not formatted
There are many unwanted fields

On friday's call we saw the load payload, it won't make sense we throw the whole dump

File: credit_card_charge_restlet/create_credit_card_charge.js (hunk 2 above)
Comment (ruuushhh) [reply credit_card_charge_restlet/create_credit_card_charge.js]: sure will try the entity and account thingy
if worked will add all else will add others

File: credit_card_charge_restlet/create_credit_card_charge.js (hunk 1 above)
Comment (ashwin1111) [reply credit_card_charge_restlet/create_credit_card_charge.js]: we're only going to do it on case by case basis - completely fine to get unwanted fields. With netsuite any data helps.

File: credit_card_charge_restlet/create_credit_card_charge.js (hunk 3 above)
Comment (ruuushhh) [reply credit_card_charge_restlet/create_credit_card_charge.js:61]: there is no selectLine method
There is selectNewLine and removeLine

That's why we remove first and add new,
 can't edit the existing one

File: credit_card_charge_restlet/create_credit_card_charge.js (hunk 2 above)
Comment (ashwin1111) [reply credit_card_charge_restlet/create_credit_card_charge.js]: sure, also handle all keys safe. In future if some other fields failed some customer we have a way to just remove those keys and send others

File: credit_card_charge_restlet/create_credit_card_charge.js (hunk 1 above)
Comment (ruuushhh) [reply credit_card_charge_restlet/create_credit_card_charge.js]: this is done

File: credit_card_charge_restlet/create_credit_card_charge.js (hunk 2 above)
Comment (ruuushhh) [reply credit_card_charge_restlet/create_credit_card_charge.js]: this is also done

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**